### PR TITLE
Add uastats, a tool to summarise user agents.

### PR DESCRIPTION
**Pull request by ryanslade**
uastats reads from stdin expecting one useragent per line. When the
input finishes it will print a summary similar to the following:

Read 319356 useragents

Browsers
BrowserChrome      183682 (57.52%)
BrowserSafari      64062 (20.06%)
BrowserIE          38184 (11.96%)
BrowserFirefox     25561 (8.00%)
BrowserSilk        3541 (1.11%)
BrowserOpera       2543 (0.80%)
BrowserAndroid     954 (0.30%)
BrowserUnknown     625 (0.20%)
BrowserBlackberry  132 (0.04%)
BrowserSpotify     69 (0.02%)
BrowserUCBrowser   3 (0.00%)

Operating Systems
OSWindows       143500 (44.93%)
OSAndroid       84930 (26.59%)
OSiOS           56519 (17.70%)
OSMacOSX        24116 (7.55%)
OSKindle        3878 (1.21%)
OSChromeOS      2804 (0.88%)
OSLinux         1629 (0.51%)
OSWindowsPhone  1206 (0.38%)
OSPlaystation   262 (0.08%)
OSUnknown       262 (0.08%)
OSXbox          125 (0.04%)
OSBlackberry    118 (0.04%)
OSNintendo      7 (0.00%)

Device Types
DeviceComputer  170833 (53.49%)
DevicePhone     114783 (35.94%)
DeviceTablet    32994 (10.33%)
DeviceConsole   393 (0.12%)
DeviceUnknown   246 (0.08%)
DeviceTV        107 (0.03%)

### CODE REVIEW

File: cmd/uastats/main.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		uasurfer.ParseUserAgent(scanner.Text(), ua)
+		stats.BrowserNames[ua.Browser.Name]++
+		stats.OSNames[ua.OS.Name]++
+		stats.DeviceTypes[ua.DeviceType]++
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Fprintln(os.Stderr, "reading standard input:", err)
+	}
+	fmt.Printf("Read %d useragents\n", count)
+	fmt.Println()
```
Comment (danbondd) [inline cmd/uastats/main.go]: Is this just used to print a blank line?

Comment (ryanslade) [reply cmd/uastats/main.go]: Yep

Comment (danbondd) [reply cmd/uastats/main.go]: Wasn't sure if you'd forgot to put some content in there ;)

**Comment by danbondd** [comment]
I think we'll also need to remove `go 1.7.x` from Travis as `sort.Slice` was only introduced in 1.8 _(I think)_

**Pull request merged**